src/components/ItemsList.js

```diff
@@ -15,16 +15,24 @@ export class ItemsList extends React.Component {
     }
   }
 
+  deleteItem (e, uid) {
+    e.preventDefault();
+    this.props.deleteItem(uid);
+    this.props.fetchAllItems();
+  }
+
   render() {
-    const { items } = this.props;
+    const { items, deleteItem } = this.props;
     return (
       <div id="items-div">
         <h2 id="items-title">My Cards</h2>
         <ul>
           { (items.data).map((item, index) => {
+              let uid = item.uid;
               return (
                 <li id="item" key={index}>
-                  <button id="delete-button">Delete</button>
+                  <button id="delete-button"
+                           onClick={e => deleteItem(e, uid)}>Delete</button>
```

Comment:
It looks like the `uid` you're passing into your action is an object that contains the event object and the `uid`. Just pass in the `uid`, you shouldn't need the event.  So it would be something like:

`onClick={e => deleteItem(uid)}`